### DISCUSSION TITLE
Values are not updating while reading from python-opcua

### DISCUSSION THREAD

**Discussion by GauravMundhra** [FreeOpcUa/python-opcua#1306] [Q&A]
Hi,
I have converted OPC-DA to OPC-UA

While reading values if the server is connected through UA Expert values are coming right. As soon as I close UA Expert the values reading through python opcua holds to previous values and does not update.

Why python opcua is depending on if UA expert is connected through server or not?

**Comment by AndreasHeine** [comment]
if you dont explain your setup no one will understand your problem...

python-opcua server dont cares if a client is connected or not!

**Reply by GauravMundhra** [reply]
Hello Sir , 

I am new to OPC servers, I have a OPC DA server which I have converted to OPC-UA. 
When I try to read values of nodes ,I get correct values only if UA Expert client is connected to server.

**Reply by AndreasHeine** [reply]
DA and UA are completly different technologies as you probably know. as i mentioned the our opcua server implementation dont care about clients connected status so ther might be another problem!
you can test some of our sample servers they should show that they run independend of a client: https://github.com/FreeOpcUa/python-opcua/tree/master/examples

you can use different client tools to verify:
UaExpert https://www.unified-automation.com/products/development-tools/uaexpert.html
Prosys OPC UA Client Browser https://downloads.prosysopc.com/opc-ua-client-downloads.php
Python OPCUA Client GUI https://github.com/FreeOpcUa/opcua-client-gui
